clang/runtime/dpct-rt/include/dpl_extras/dpcpp_extensions.h.inc

```diff
@@ -554,6 +554,31 @@ reduce(Item item, T (&inputs)[VALUES_PER_THREAD], BinaryOperation binary_op) {
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|reduce_over_partial_group|dpct::group
+// DPCT_DEPENDENCY_BEGIN
+// DplExtrasDpcppExtensions|group_reduce
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+/// Perform a reduction on a limited number of the work items in a subgroup
+///
+/// \param item A work-item in a group.
+/// \param value value per work item which is to be reduced
+/// \param items_to_reduce num work items at the start of the subgroup to reduce
+/// \param identity identity for type and binary_op
+/// \param binary_op functor that implements the binary operation used to
+/// perform the scan. \returns value of the reduction using binary_op
+template <typename Item, typename T, class BinaryOperation>
+__attribute__((always_inline)) T
```

Comment:
Can we use `inline` here? `_attribute__((always_inline))` is a GCC/Clang extension

Reply:
Sure,  I was just following the convention of the surrounding functions.  Should we adjust those as well?

Reply:
Ah, let's still following the convention of the surrounding functions.